src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int MAX_SIDE_LENGTH = 100;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        int firstSide = random.nextInt(MAX_SIDE_LENGTH);
+        int secondSide = random.nextInt(MAX_SIDE_LENGTH);
+        int thirdSide = random.nextInt(MAX_SIDE_LENGTH);
```

Comment:
what if it will be a `Circle`? We'll need only one side, but we will still create three more

Reply:
optional sides are now created only in case where they are needed